### DISCUSSION TITLE
Using with Next.js 13

### DISCUSSION THREAD

**Discussion by ziaukhan** [chakra-ui/chakra-ui#6908] [General]
I am trying to use Chakra UI with Next.js 13 but getting errors. Is there any documentation available?

**Comment by viczam** [comment]
emotion is not supported yet on nextjs 13

**Comment by cmaciasjimenez** [comment]
I was going to ask the same question. Just tried a new project with Next.js 13 and can't seem to run Chakra with the new `app` directory. CSS doesn't seem to load and I'm getting tons of errors. I don't see any documentation on the ChakraUI site or Docs sections (I think at least an acknowledgment of Next.js 13 with breaking changes should be posted in the Next.js guide. Don't even know if I'm supposed to use (or how to) the `ChakraProvider` with page.jsx, etc.

Are we supposed to keep using the old `pages` directory for the time being?

Any guidance or help is much appreciated.

Thank you

**Reply by dvartic** [reply]
I am starting to work on the frontend of a big project now and decided to start using Next13 beta features to avoid using patterns such as getServerSideProps. I also run into the same issue. To enable support for useContext hooks in the Chakra Provider use this reference: https://beta.nextjs.org/docs/rendering/server-and-client-components#third-party-packages

You need to create another file for providers with `'use client';` directive, then import that component into `layout.tsx`

@viczam Could you please elaborate on what it is not supported yet?

I ran into an issue with the new next/link, that renders an `<a>` I am unsure how to style that using Chakra Link.

**Reply by mheob** [reply]
> I ran into an issue with the new next/link, that renders an `<a>` I am unsure how to style that using Chakra Link.

Maybe https://github.com/chakra-ui/chakra-ui/pull/6912#issuecomment-1293879562 helps.

**Reply by dvartic** [reply]
> > I ran into an issue with the new next/link, that renders an `<a>` I am unsure how to style that using Chakra Link.
> 
> Maybe [#6912 (comment)](https://github.com/chakra-ui/chakra-ui/pull/6912#issuecomment-1293879562) helps.

Thank you, it does work. I did not know how to use the `as` prop. I was using NextLink with `legacyBehavior` and `passHref` which works too but this way we can use the standard behavior.

**Comment by dvartic** [comment]
As far as I understand all libraries work ok including Chakra as long as they are used in Client components (since Chakra depends on Context for example).

I did run into issues with the new DataStreaming/Suspense that React18 and Next13 enable with Server Components.

I am trying to get data from DB using Prisma. This data is needed to render a Client Component. I am unable to Suspense rendering of this Client Comp until this data is available. Passing promises from Server Comp to Client Comp is not supported. Prisma Client cannot be used on client obviously.

I assume possible workaround is to create an API route and use the new React18 `fetch()` and `use()`

**Comment by davidsonsns** [comment]
Aight, answering the question through code:

app/layout.tsx
```tsx
import Providers from './providers'

const RootLayout = ({ children }: { children: React.ReactNode }) => {
  return (
    <html>
      <body>
		<Providers>{children}</Providers>
	 </body>
    </html>
  )
}

export default RootLayout
```

app/providers.tsx
```tsx
'use client' // ---> this line does the trick

import { ChakraProvider } from '@chakra-ui/react'
import { ReactNode } from 'react'
import theme from 'src/ui/theme'

const Providers = ({ children }: { children: ReactNode }) => {
  return (
      <ChakraProvider resetCSS theme={theme}>
        {children}
      </ChakraProvider>
  )
}

export default Providers
```

reference: https://beta.nextjs.org/docs/rendering/server-and-client-components#context

**Reply by shinebayar-g** [reply]
Thanks for the code. Hope this lands on https://chakra-ui.com/getting-started/nextjs-guide#nextjs-13 documentation. While it mentions the use of `'use client';`, it doesn't mention how to setup Provider in code.

**Reply by venkata-penumatsa** [reply]
This code snippet should, indeed, be included in the documentation. As a newcomer, it's difficult to get started with so little information.

**Reply by ODelibalta** [reply]
This works until I add colormodescript to have color modes.

**Reply by bogdan-marian** [reply]
I'm new. Please provide a code also for `/src/app/page.js` I have no ideea what to do here.

**Comment by fetimo** [comment]
Are there any plans to make Chakra UI compatible with Server Components without `'use client'` or are they incompatible?

**Reply by baauer90** [reply]
Any update on this?

**Comment by SheaBelsky** [comment]
I'm struggling to get this to work on a light mode/dark mode application. Where is `ColorModeScript` meant to go in relation to other files in the new Next.js 13 hierarchy? @davidsonsns posted a great start, but I always get a significant color mode "blink" from light to black whenever my app is meant to start in Dark Mode. What's the reason for this?

Current code: https://github.com/SheaBelsky/SheaWebsite2/blob/5c55a356940c928b11d62cd101fd32b99069f80c/app/providers.tsx#L14

**Reply by danilocecilia** [reply]
Were you able to solve this issue ? I'm having the same problem.

**Reply by danilocecilia** [reply]
OK, I've figure it out.

After updating these 2 packages to these specified version(latest) it started working: 
`"@chakra-ui/next-js": "^2.0.1",
    "@chakra-ui/react": "^2.5.2",`

**Comment by curiouscod3** [comment]
Hope this is helpful 

https://github.com/vercel/next.js/discussions/43802#discussioncomment-4356708

**Comment by ODelibalta** [comment]
@SheaBelsky same here. Color mode script does not work for me. My office paid for a pro theme and when we try to use that, there is a FOUC issue. Been trying to get around that issue for a while now, its quite frustrating.
